include/jemalloc/internal/mutex.h

```diff
@@ -216,7 +221,6 @@ malloc_mutex_lock(tsdn_t *tsdn, malloc_mutex_t *mutex) {
 	if (isthreaded) {
 		if (malloc_mutex_trylock_final(mutex)) {
 			malloc_mutex_lock_slow(mutex);
-			atomic_store_b(&mutex->locked, true, ATOMIC_RELAXED);
```

Comment:
Good catch! Maybe comment the idea that "the thread holding the lock should be the one updating `mutex->locked`" somewhere to prevent future mistakes on updating it? Its definition [here](https://github.com/jemalloc/jemalloc/blob/db4f0e71820017039f09e5acc04b554826e304fd/include/jemalloc/internal/mutex.h#L40) is a good candidate.

Reply:
Added comments there.